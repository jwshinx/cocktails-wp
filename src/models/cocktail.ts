interface Thumbable {
  strDrinkThumb: string
}
export interface GeneralDrink extends Thumbable {
  idDrink: string
  strDrink: string
}

export interface PropertyType<T> {
  property: keyof T
  isDescending: boolean
}
export interface CocktailFilter<T> {
  property: keyof T
  isTruthySelected: boolean
}

export type FilterSegmentType = 'filter'
export type SearchSegmentType = 'search'

export type SegmentType = FilterSegmentType | SearchSegmentType

export interface SegmentKeyType {
  readonly name: string
  readonly value: string
  readonly label: string
}

export const AlcoholicSegmentKey: SegmentKeyType = {
  label: 'Alcoholic / Nonalcoholic',
  name: 'alcoholic',
  value: 'a',
}
export const IngredientSegmentKey: SegmentKeyType = {
  label: 'Ingredient',
  name: 'ingredient',
  value: 'i',
}
export const FirstLetterSegmentKey: SegmentKeyType = {
  label: 'First Letter',
  name: 'firstLetter',
  value: 'f',
}
export const CategorySegmentKey: SegmentKeyType = {
  label: 'Category',
  name: 'category',
  value: 'c',
}
export const CocktailNameSegmentKey: SegmentKeyType = {
  label: 'Cocktail Name',
  name: 'cocktailName',
  value: 's',
}
export const GlassSegmentKey: SegmentKeyType = {
  label: 'Glass',
  name: 'glass',
  value: 'g',
}

export interface OptionValueType {
  readonly label: string
  readonly value: string
}

export const AlcoholicValue: OptionValueType = {
  label: 'Alcoholic',
  value: 'alcoholic',
}
export const NonAlcoholicValue: OptionValueType = {
  label: 'Non Alcoholic',
  value: 'non_alcoholic',
}
export const OptionalAlcoholValue: OptionValueType = {
  label: 'Optional Alcohol',
  value: 'optional_alcohol',
}

export const AlcoholicFilterOptions = [
  AlcoholicValue,
  NonAlcoholicValue,
  OptionalAlcoholValue,
]

export const OrdinaryDrinkValue: OptionValueType = {
  label: 'Ordinary Drink',
  value: 'Ordinary Drink',
}
export const CocktailValue: OptionValueType = {
  label: 'Cocktail',
  value: 'Cocktail',
}
export const MilkFloatShakeValue: OptionValueType = {
  label: 'Milk / Float / Shake',
  value: 'Milk / Float / Shake',
}
export const OtherUnknownValue: OptionValueType = {
  label: 'Other / Unknown',
  value: 'Other/Unknown',
}
export const CocoaValue: OptionValueType = { label: 'Cocoa', value: 'Cocoa' }
export const ShotValue: OptionValueType = { label: 'Shot', value: 'Shot' }
export const CoffeeTeaValue: OptionValueType = {
  label: 'Coffee / Tea',
  value: 'Coffee / Tea',
}
export const HomemadeLiquerValue: OptionValueType = {
  label: 'Homemade Liqueur',
  value: 'Homemade Liqueur',
}
export const PunchPartyDrinkValue: OptionValueType = {
  label: 'Punch / Party Drink',
  value: 'Punch / Party Drink',
}
export const BeerValue: OptionValueType = { label: 'Beer', value: 'Beer' }
export const SoftDrinkSodaValue: OptionValueType = {
  label: 'Soft Drink / Soda',
  value: 'Soft Drink / Soda',
}

export const CategoryFilterOptions = [
  OrdinaryDrinkValue,
  CocktailValue,
  MilkFloatShakeValue,
  OtherUnknownValue,
  CocoaValue,
  ShotValue,
  CoffeeTeaValue,
  HomemadeLiquerValue,
  PunchPartyDrinkValue,
  BeerValue,
  SoftDrinkSodaValue,
]

export const HighballGlassValue: OptionValueType = {
  label: 'Highball glass',
  value: 'Highball glass',
}
export const CocktailGlassValue: OptionValueType = {
  label: 'Cocktail glass',
  value: 'Cocktail glass',
}
export const OldFashionedGlassValue: OptionValueType = {
  label: 'Old-fashioned glass',
  value: 'Old-fashioned glass',
}
export const WhiskeyGlassValue: OptionValueType = {
  label: 'Whiskey Glass',
  value: 'Whiskey Glass',
}
export const CollinsGlassValue: OptionValueType = {
  label: 'Collins glass',
  value: 'Collins glass',
}
export const PousseCafeGlassValue: OptionValueType = {
  label: 'Pousse cafe glass',
  value: 'Pousse cafe glass',
}
export const ChampagneFluteValue: OptionValueType = {
  label: 'Champagne flute',
  value: 'Champagne flute',
}
export const WhiskeySourGlassValue: OptionValueType = {
  label: 'Whiskey sour glass',
  value: 'Whiskey sour glass',
}
export const CordialGlassValue: OptionValueType = {
  label: 'Cordial glass',
  value: 'Cordial glass',
}
export const BrandySnifterValue: OptionValueType = {
  label: 'Brandy snifter',
  value: 'Brandy snifter',
}
export const WhiteWineGlassValue: OptionValueType = {
  label: 'White wine glass',
  value: 'White wine glass',
}
export const NickAndNorGlassValue: OptionValueType = {
  label: 'Nick and Nora Glass',
  value: 'Nick and Nora Glass',
}
export const HurricaneGlassValue: OptionValueType = {
  label: 'Hurricane glass',
  value: 'Hurricane glass',
}
export const CoffeeMugGlassValue: OptionValueType = {
  label: 'Coffee mug',
  value: 'Coffee mug',
}
export const ShotGlassValue: OptionValueType = {
  label: 'Shot glass',
  value: 'Shot glass',
}
export const JarGlassValue: OptionValueType = { label: 'Jar', value: 'Jar' }
export const IrishCoffeeCupGlassValue: OptionValueType = {
  label: 'Irish coffee cup',
  value: 'Irish coffee cup',
}
export const PunchBowlGlassValue: OptionValueType = {
  label: 'Punch bowl',
  value: 'Punch bowl',
}
export const PitcherGlassValue: OptionValueType = {
  label: 'Pitcher',
  value: 'Pitcher',
}
export const PintGlassValue: OptionValueType = {
  label: 'Pint glass',
  value: 'Pint glass',
}
export const CopperMugGlassValue: OptionValueType = {
  label: 'Copper Mug',
  value: 'Copper Mug',
}
export const WineGlassValue: OptionValueType = {
  label: 'Wine Glass',
  value: 'Wine Glass',
}
export const BeerMugGlassValue: OptionValueType = {
  label: 'Beer mug',
  value: 'Beer mug',
}
export const MargaritaCoupetteGlassValue: OptionValueType = {
  label: 'Margarita/Coupette glass',
  value: 'Margarita/Coupette glass',
}
export const BeerPilsnerGlassValue: OptionValueType = {
  label: 'Beer pilsner',
  value: 'Beer pilsner',
}
export const BeerGlassValue: OptionValueType = {
  label: 'Beer Glass',
  value: 'Beer Glass',
}
export const ParfaitGlassValue: OptionValueType = {
  label: 'Parfait glass',
  value: 'Parfait glass',
}
export const MasonJarGlassValue: OptionValueType = {
  label: 'Mason jar',
  value: 'Mason jar',
}
export const MargaritaGlassValue: OptionValueType = {
  label: 'Margarita glass',
  value: 'Margarita glass',
}
export const MartiniGlassValue: OptionValueType = {
  label: 'Martini Glass',
  value: 'Martini Glass',
}
export const BalloonGlassValue: OptionValueType = {
  label: 'Balloon Glass',
  value: 'Balloon Glass',
}
export const CoupeGlassValue: OptionValueType = {
  label: 'Coupe Glass',
  value: 'Coupe Glass',
}

export const GlassFilterOptions = [
  HighballGlassValue,
  CocktailGlassValue,
  OldFashionedGlassValue,
  WhiskeyGlassValue,
  CollinsGlassValue,
  PousseCafeGlassValue,
  ChampagneFluteValue,
  WhiskeySourGlassValue,
  CordialGlassValue,
  BrandySnifterValue,
  WhiteWineGlassValue,
  NickAndNorGlassValue,
  HurricaneGlassValue,
  CoffeeMugGlassValue,
  ShotGlassValue,
  JarGlassValue,
  IrishCoffeeCupGlassValue,
  PunchBowlGlassValue,
  PitcherGlassValue,
  PintGlassValue,
  CopperMugGlassValue,
  WineGlassValue,
  BeerMugGlassValue,
  MargaritaCoupetteGlassValue,
  BeerPilsnerGlassValue,
  BeerGlassValue,
  ParfaitGlassValue,
  MasonJarGlassValue,
  MargaritaGlassValue,
  MartiniGlassValue,
  BalloonGlassValue,
  CoupeGlassValue,
]

export interface DetailDrink extends GeneralDrink {
  dateModified: string
  idDrink: string
  strAlcoholic: string // "Alcoholic",
  strCategory: string // "Ordinary Drink",
  strCreativeCommonsConfirmed: string // "No",
  strDrink: string // "Tuxedo Cocktail",
  strDrinkAlternate: string | null // null,
  strDrinkThumb: string // "https://www.thecocktaildb.com/images/media/drink/4u0nbl1504352551.jpg",
  strGlass: string // "Cocktail glass",
  strIBA: string // "Unforgettables",
  strImageAttribution: string | null // null,
  strImageSource: string | null // null,
  strIngredient1: string | null // "Dry Vermouth",
  strIngredient2: string | null // "Gin",
  strIngredient3: string | null // "Maraschino liqueur",
  strIngredient4: string | null // "Anis",
  strIngredient5: string | null // "Orange bitters",
  strIngredient6: string | null // "Cherry",
  strIngredient7: string | null // null,
  strIngredient8: string | null // null,
  strIngredient9: string | null // null,
  strIngredient10: string | null // null,
  strIngredient11: string | null // null,
  strIngredient12: string | null // null,
  strIngredient13: string | null // null,
  strIngredient14: string | null // null,
  strIngredient15: string | null // null,
  strInstructions: string | null // "Stir all ingredients with ice and strain into a cocktail glass. Garnish with a cherry and a twist of lemon zest.",
  strInstructionsDE: string | null // "Alle Zutaten mit Eis verr√ºhren und in ein Cocktailglas abseihen. Mit einer Kirsche und einem Schuss Zitrone garnieren.",
  strInstructionsES: string | null // null,
  strInstructionsFR: string | null // null,
  strInstructionsIT: string | null // "Mescolare tutti gli ingredienti con ghiaccio e filtrare in una coppetta da cocktail. Guarnire con una ciliegia e una scorza di limone grattugiata.",
  'strInstructionsZH-HANS': string | null // null,
  'strInstructionsZH-HANT': string | null // null,
  strMeasure1: string | null // "1 1/2 oz ",
  strMeasure2: string | null // "1 1/2 oz ",
  strMeasure3: string | null // "1/4 tsp ",
  strMeasure4: string | null // "1/4 tsp ",
  strMeasure5: string | null // "2 dashes ",
  strMeasure6: string | null // "1 ",
  strMeasure7: string | null // null,
  strMeasure8: string | null // null,
  strMeasure9: string | null // null,
  strMeasure10: string | null // null,
  strMeasure11: string | null // null,
  strMeasure12: string | null // null,
  strMeasure13: string | null // null,
  strMeasure14: string | null // null,
  strMeasure15: string | null // null,
  strTags: string | null // "IBA,Classic",
  strVideo: string | null // "https://www.youtube.com/watch?v=347ZY5eJFU8",
}

export interface ApiRawCocktailData {
  drinks: Array<DetailDrink>
}
