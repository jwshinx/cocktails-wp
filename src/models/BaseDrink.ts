import { GeneralDrink } from '../models/cocktail'

class BaseDrink {
  idDrink: string
  strDrink: string
  strDrinkThumb: string

  constructor(item: GeneralDrink) {
    this.strDrinkThumb = item.strDrinkThumb
    this.idDrink = item.idDrink
    this.strDrink = item.strDrink
  }

  getImageUrl(): string {
    return this.strDrinkThumb
  }

  getSummary(): string {
    return `${this.idDrink}: ${this.strDrink}`
  }

  getInfo(): Array<string> {
    return [this.idDrink, this.strDrink]
  }
}

export default BaseDrink
