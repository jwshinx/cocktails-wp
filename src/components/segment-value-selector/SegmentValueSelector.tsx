import React from 'react'
import SegmentValueSelectorOptions from './SegmentValueSelectorOptions'
import { OptionValueType } from '../../models/cocktail'

export default function SegmentValueSelector({
  items,
  onSegmentValueChange,
}: {
  items: Array<OptionValueType>
  onSegmentValueChange: (value: string) => void
}): JSX.Element {
  const onValueSelectorChange = (event: any): void => {
    onSegmentValueChange(event.target.value as string)
  }

  return (
    <div className="form-group">
      <label htmlFor="segment-value">Enter variety</label>
      <select
        className="form-select mb-3"
        id="segment-value"
        onChange={(event) => onValueSelectorChange(event)}
      >
        <SegmentValueSelectorOptions items={items} />
      </select>
    </div>
  )
}
