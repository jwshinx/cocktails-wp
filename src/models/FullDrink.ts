import _ from 'lodash'

import BaseDrink from '../models/BaseDrink'
import { DetailDrink } from './cocktail'

class FullDrink extends BaseDrink {
  ingredientCount: number
  allIngredients: Array<string>

  constructor(item: DetailDrink) {
    super(item)
    this.ingredientCount = 0

    this.strDrinkThumb = item.strDrinkThumb
    this.idDrink = item.idDrink
    this.strDrink = item.strDrink

    this.allIngredients = []
    if (item.strIngredient1) {
      this.allIngredients.push(item.strIngredient1)
    }
    if (item.strIngredient2) {
      this.allIngredients.push(item.strIngredient2)
    }
    if (item.strIngredient3) {
      this.allIngredients.push(item.strIngredient3)
    }
    if (item.strIngredient4) {
      this.allIngredients.push(item.strIngredient4)
    }
    if (item.strIngredient5) {
      this.allIngredients.push(item.strIngredient5)
    }
    if (item.strIngredient6) {
      this.allIngredients.push(item.strIngredient6)
    }
    if (item.strIngredient7) {
      this.allIngredients.push(item.strIngredient7)
    }
    if (item.strIngredient8) {
      this.allIngredients.push(item.strIngredient8)
    }
    if (item.strIngredient9) {
      this.allIngredients.push(item.strIngredient9)
    }
    if (item.strIngredient10) {
      this.allIngredients.push(item.strIngredient10)
    }
    if (item.strIngredient11) {
      this.allIngredients.push(item.strIngredient11)
    }
    if (item.strIngredient12) {
      this.allIngredients.push(item.strIngredient12)
    }
    if (item.strIngredient13) {
      this.allIngredients.push(item.strIngredient13)
    }
    if (item.strIngredient14) {
      this.allIngredients.push(item.strIngredient14)
    }
    if (item.strIngredient15) {
      this.allIngredients.push(item.strIngredient15)
    }
  }

  getList(): string {
    return _.join(this.allIngredients, ', ')
  }

  getDifficulty(): string {
    if (this.allIngredients.length < 4) {
      return 'Easy'
    } else if (
      this.allIngredients.length === 4 ||
      this.allIngredients.length === 5
    ) {
      return 'Moderate'
    } else {
      return 'Difficult'
    }
  }
}

export default FullDrink
