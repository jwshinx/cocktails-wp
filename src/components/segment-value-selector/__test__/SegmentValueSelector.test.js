/**
 * @jest-environment jsdom
 */

import React from 'react'
// import ReactDOM from 'react-dom'
// import { render } from '@testing-library/react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

import SegmentValueSelector from '../SegmentValueSelector'
import { AlcoholicFilterOptions } from '../../../models/cocktail'

it('should render label "SegmentValueSelector"', () => {
  render(<SegmentValueSelector items={AlcoholicFilterOptions} />)
  const elem = screen.getByText(/SegmentValueSelector/)
  expect(elem).toBeInTheDocument()
})

it('should render combobox', () => {
  render(<SegmentValueSelector items={AlcoholicFilterOptions} />)
  const elem = screen.getByRole('combobox')
  expect(elem).toBeInTheDocument()
})

it('should render options "filter" and "search"', () => {
  render(<SegmentValueSelector items={AlcoholicFilterOptions} />)
  expect(screen.getAllByRole('option', { name: 'Alcoholic' })).toBeTruthy()
  expect(screen.getAllByRole('option', { name: 'Non Alcoholic' })).toBeTruthy()
  expect(
    screen.getAllByRole('option', { name: 'Optional Alcohol' })
  ).toBeTruthy()
})

it('should snapshot SegmentValueSelector', () => {
  const tree = renderer
    .create(<SegmentValueSelector items={AlcoholicFilterOptions} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
