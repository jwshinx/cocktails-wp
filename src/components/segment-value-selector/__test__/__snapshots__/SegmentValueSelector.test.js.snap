// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should snapshot SegmentValueSelector 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="segment-value"
  >
    SegmentValueSelector
  </label>
  <select
    className="form-select mb-3"
    id="segment-value"
    onChange={[Function]}
  >
    <option
      value="alcoholic"
    >
      Alcoholic
    </option>
    <option
      value="non_alcoholic"
    >
      Non Alcoholic
    </option>
    <option
      value="optional_alcohol"
    >
      Optional Alcohol
    </option>
  </select>
</div>
`;
